### ISSUE TITLE
upgrade guide from cybercog/laravel-likeable ?

### ISSUE THREAD

**Issue by vesper8** [cybercog/laravel-love#8]
I've been using cybercog/laravel-likeable for what seems like a solid year.. in a production environment too

Now I see you're putting out this new and improved version of it.. but I am wondering if the upgrade is super straightforward or not? I obviously don't want to lose any of the existing data

any chance you could post a step-by-step upgrade guide since you know best what's changed and what backwards compatibility might be affected?

Also, I had asked you long ago about the possibility of using cybercog/laravel-likeable to use dynamic states

for example.. by default this package provides the ability for a model to like/dislike another model

I want that, and I am using that. But I also need to allow models to do things like "block/unblock" models and "hide/unhide" models

It would be great if I could do all of these things with this package alone.. does your new version attempt to cover this use case by any chance?

many thanks!

**Comment by a-komarev** [comment]
Hello @vesper8! Thank you for kind words. I'm glad that it's doing the job for you.

All upgrade is described in [Upgrade guide from v3 to v4](https://github.com/cybercog/laravel-love/blob/master/UPGRADING.md#from-v3-to-v4) since it's just a fork from Laravel Likeable which had v3 release as latest one. I've tried to describe upgrade process as much detailed as I can. Then you need to upgrade from v4 to v5 (it is described in Upgrade Guide too). If there are any issues with it, feel free to push PR which will introduce a fix.

Dynamic states is very interesting thing but it's not great to mix different kind of data in one database table because on big data volumes you will have an issues with fetching records. I'd recommend to add this functionality with additional packages. I'm trying to design package to do only one kind of tasks at a time.

For block\unblock I'm using [Laravel Ban package](https://github.com/cybercog/laravel-ban). One of the reasons why I recommend to split it to different packages is requirement to store all block history in separate database table. This could be useful for audits for example or to have extra logic (when have more than 5 blocks in history for this records - permanent ban it). But if you are talking about blocking content for concrete user only and not globally - it's not that easy to make such things abstract and make it work in any project out of the box without code coupling and good performance.

Same for hide\unhide functionality. I have closed source solution, but it's very complex and tightly coupled with project where it's used, otherwise queries will be too slow. I don't see any optimal way to do it, and thats why I didn't published it yet and there are no future plans for it.

In future there are plans to add my package PHP Eumerated as dependency to Laravel Love, but I need to open-source it first and drop support of PHP < 7.1. It will give us a chance to overwrite and extend Like types on application level. So you will be able to introduce your own Like types.

Laravel Love v6 is planned to be released after Laravel 5.6 being published, because it will use latest relationships features to optimize queries and perform database refactoring to make it less fragile.

**Comment by a-komarev** [comment]
I'm closing this issue. Feel free to continue conversation here or open new issue if there are any upgrade questions will appear.
